### PR TITLE
fix: prevent nil pointer panic because of nonexist attribute of Access Log

### LINKED ISSUE #1830: EnvoyGateway crashes (nil pointer) due to inappropriate AccessLog settings
*Description*:
Envoy Gateway crashes (nil pointer) when defining `EnvoyProxy` with `AccessLog` enabled using the `File` sink type. It happened when we didn't provide the `file` content under `spec.telemetry.accessLog.settings.sinks`.

*Repro steps*:
Apply below `EnvoyProxy` manifest:
```yaml
apiVersion: config.gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  name: default
  namespace: envoy-gateway-system
spec:
  telemetry:
    accessLog:
      settings:
      - format:
          json:
            host: '%REQ(:AUTHORITY)%'
          type: JSON
        sinks:
        - type: File
```

*Environment*:
Envoy Gateway v0.5.0 and latest

*Logs*:
```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d668af]

goroutine 72 [running]:
github.com/envoyproxy/gateway/internal/gatewayapi.processAccessLog(0xc001218780)
        /home/runner/work/gateway/gateway/internal/gatewayapi/listener.go:193 +0x28f
github.com/envoyproxy/gateway/internal/gatewayapi.(*Translator).ProcessListeners(0xc0008a1bf0?, {0xc0000128c8?, 0x1, 0x1}, 0x3c00900?, 0x55?, 0xc000bb1500)
        /home/runner/work/gateway/gateway/internal/gatewayapi/listener.go:52 +0x5fe
github.com/envoyproxy/gateway/internal/gatewayapi.(*Translator).Translate(0xc000083fc0?, 0xc000bb1500)
        /home/runner/work/gateway/gateway/internal/gatewayapi/translator.go:121 +0xbd
github.com/envoyproxy/gateway/internal/gatewayapi/runner.(*Runner).subscribeAndTranslate.func1({{0xc000080e40?, 0x16?}, 0x0?, 0xc000bb1500?})
        /home/runner/work/gateway/gateway/internal/gatewayapi/runner/runner.go:78 +0x16e
github.com/envoyproxy/gateway/internal/message.HandleSubscription[...](0x2769c40, 0xc0008a1f98?)
        /home/runner/work/gateway/gateway/internal/message/watchutil.go:36 +0x148
github.com/envoyproxy/gateway/internal/gatewayapi/runner.(*Runner).subscribeAndTranslate(0xc0000f5280, {0x2749548?, 0xc00013c5f0?})
        /home/runner/work/gateway/gateway/internal/gatewayapi/runner/runner.go:52 +0x75
created by github.com/envoyproxy/gateway/internal/gatewayapi/runner.(*Runner).Start
        /home/runner/work/gateway/gateway/internal/gatewayapi/runner/runner.go:46 +0x272
```

### PR DESCRIPTION
**Pull request by ardikabs**
Fixes #1830 
Additionally, I also added a test for JSON format type Access Log.

### CODE REVIEW

**Comment by Xunzhuo** [comment]
Plz run `make testdata` and submit again, thanks.

File: internal/gatewayapi/listener.go [hunk 1]
```diff
@@ -175,6 +175,10 @@ func processAccessLog(envoyproxy *configv1a1.EnvoyProxy) *ir.AccessLog {
 		for _, sink := range accessLog.Sinks {
 			switch sink.Type {
```
Comment (Xunzhuo) [inline internal/gatewayapi/listener.go:176]: Do we need a default in switch to deal with the unsupported sink type? @zirain

Comment (zirain) [reply internal/gatewayapi/listener.go:176]: that will block by kube-apiserver?

Comment (ardikabs) [reply internal/gatewayapi/listener.go:176]: ya, I think that was already prevented from the schema validation by kube-apiserver, as `sink.Type` is actually an enum of `File` and `OpenTelemetry`.

**Comment by arkodg** [comment]
imo this should be handled in https://github.com/envoyproxy/gateway/blob/3999cdd3604ad6f946a7cba795f2b034a1ae0e22/api/config/v1alpha1/validation/validate.go#L40 and we must error out early if `type: File` has a `nil` , `file` field, this will be caught in the provider layer which reconciles the resource and the error will surface in the `GatewayClass` status condition

File: internal/gatewayapi/listener.go (hunk 1 above)
Comment (Xunzhuo) [reply internal/gatewayapi/listener.go:176]: Right, thanks.

**Comment by ardikabs** [comment]
> imo this should be handled in
> 
> https://github.com/envoyproxy/gateway/blob/3999cdd3604ad6f946a7cba795f2b034a1ae0e22/api/config/v1alpha1/validation/validate.go#L40
> 
> and we must error out early if `type: File` has a `nil` , `file` field, this will be caught in the provider layer which reconciles the resource and the error will surface in the `GatewayClass` status condition

but @arkodg , as per the [docs](https://gateway.envoyproxy.io/latest/api/config_types.html#fileenvoyproxyaccesslog), it said, it is expected to disable the access log for `type: File` if path was not defined or empty. So, it seems to me adding an error to the gatewayclass status would against with the docs, wdyt?

**Comment by arkodg** [comment]
oh good catch @ardikabs , I think that is a docs error. we have a field called `disable` to disable access logs now, let me fix that https://gateway.envoyproxy.io/v0.5.0/api/config_types.html#proxyaccesslog

**Comment by arkodg** [comment]
raised https://github.com/envoyproxy/gateway/pull/1850 to fix doc string discrepancy.

File: api/config/v1alpha1/validation/validate_test.go [hunk 2]
```diff
@@ -246,12 +246,65 @@ func TestValidateEnvoyProxy(t *testing.T) {
...
+			},
+			expected: false,
+		},
 	}
 
 	for i := range testCases {
 		tc := testCases[i]
 		t.Run(tc.name, func(t *testing.T) {
 			err := ValidateEnvoyProxy(tc.proxy)
+			t.Log(err)
```
Comment (Xunzhuo) [inline api/config/v1alpha1/validation/validate_test.go:307]: Is this required?

Comment (ardikabs) [reply api/config/v1alpha1/validation/validate_test.go:307]: ah my bad @Xunzhuo, let me remove it.

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**